packages/expo-updates/android/src/main/java/expo/modules/updates/loader/EmbeddedLoader.java

```diff
@@ -164,8 +196,26 @@ private void copyAllAssets(ArrayList<AssetEntity> assetList) {
     }
     mDatabase.assetDao().insertAssets(mFinishedAssetList, mUpdateEntity);
     if (mErroredAssetList.size() == 0) {
-      mDatabase.updateDao().markUpdateFinished(mUpdateEntity);
+      mDatabase.updateDao().markUpdateFinished(mUpdateEntity, mSkippedAssetList.size() != 0);
     }
     // TODO: maybe try downloading failed assets in background
   }
+
+  private boolean shouldSkipAsset(AssetEntity asset) {
+    if (asset.scales == null || asset.scale == null) {
+      return false;
+    }
+    return pickClosestScale(asset.scales) != asset.scale;
+  }
+
+  // https://developer.android.com/guide/topics/resources/providing-resources.html#BestMatch
+  private float pickClosestScale(Float[] scales) {
```

Comment:
Are we guaranteed the scales are sorted? Also this should start from the highest scale and go down to the lowest so we pick the closest scale that is greater than or equal to the actual screen scale. From that link:
> In general, Android prefers scaling down a larger original image to scaling up a smaller original image

Reply:
Thanks for pointing this out -- I realized I had been looking at the wrong section on that link (choosing screen size rather than screen density) and the logic was different 😱 I've updated this function to match Android's algorithm for picking resources based on screen density and tested it with emulators of different screen density.